### PR TITLE
Add test for preview view

### LINKED ISSUE #285: Write test for previews view

### PR DESCRIPTION
**Pull request by Le6ow5k1**
Fixes #285

### CODE REVIEW

**Comment by joshsmith** [comment]
@Le6ow5k1 thanks for the PR! Tests are failing for reasons unrelated to your pull. Forks don't have access to our ENV.

In `circle.yml`, can you change:

``` yaml
test:
  override:
    - mix test --include requires_env --cover
```

to

``` yaml
test:
  override:
    - MIX_ENV=test mix test --include requires_env --cover
```

Push that as a separate commit, if you would? Thanks!

**Comment by joshsmith** [comment]
Actually, hold tight on this suggestion for a moment.

**Comment by Le6ow5k1** [comment]
Sorry, already pushed. It seems that it needs a complete set of env variables.

**Comment by joshsmith** [comment]
@Le6ow5k1 it's running the `requires_env` tags, which you don't have on a fork for security reasons in CI.

I'm going to need to circle back to this unfortunately. Today's my wife and my anniversary.

I do think the issue can be resolved not by removing `requires_env` (we _really_ need that on non-fork CIs), but maybe by running some Circle config that works differently on forks.

**Comment by joshsmith** [comment]
We could run an if/else check on `CIRCLE_PR_USERNAME` in the `test` command so that we _only_ run the command if we assume you have the environment (it's not a fork).

**Comment by joshsmith** [comment]
@Le6ow5k1 looks like you fixed it! Apologies for the difficulties there. A new package changed some behavior along with our recent change to run those require_envs on CI. We hadn't a fork push since so it wasn't reflected.

File: circle.yml [hunk 1]
```diff
@@ -13,7 +13,10 @@ dependencies:
     - mix compile
 test:
   override:
-    - MIX_ENV=test mix test --cover
+    - |
+      if [ ${CIRCLE_PR_USERNAME} ]; then MIX_ENV=test mix test --cover; else MIX_ENV=test mix test --include requires_env --cover; fi
```
Comment (sbatson5) [inline circle.yml:17]: 👏 Nice solution! Could we get this as a separate PR?

File: test/support/factories.ex [hunk 2]
```diff
@@ -128,4 +128,12 @@ defmodule CodeCorps.Factories do
       skill: build(:skill)
     }
   end
+
+  def preview_factory do
+    %CodeCorps.Preview{
+      body: "Foo",
+      markdown: "Bar",
```
Comment (joshsmith) [inline test/support/factories.ex:135]: For this part I would make the markdown and body match up. Also, does the factory need the body of it has the markdown? There seem to be inconsistencies with other factories.

File: test/views/preview_view_test.exs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    expected_json = %{
+      data: %{
+        id: preview.id |> Integer.to_string,
+        type: "preview",
+        attributes: %{
+          "markdown" => preview.markdown,
+          "body" => preview.body,
+          "inserted-at" => preview.inserted_at,
+          "updated-at" => preview.updated_at
```
Comment (joshsmith) [inline test/views/preview_view_test.exs:26]: Would alpha order these.  Need to add this to a style guide.

File: circle.yml [hunk 4]
```diff
@@ -13,7 +13,10 @@ dependencies:
     - mix compile
 test:
   override:
-    - mix test --include requires_env --cover
+    - |
+      if [ ${CIRCLE_PR_USERNAME} ]; then MIX_ENV=test mix test --cover; else MIX_ENV=test mix test --include requires_env --cover; fi
```
Comment (joshsmith) [inline circle.yml:17]: This looks great. Can we add a line to docs explaining the change? Think there's à CI section.

**Review by joshsmith — CHANGES_REQUESTED** [review]
Minor changes here. After those will accept and ask you to rebase and squash after that. Then we'll merge!

🎈 Thanks for this.

File: circle.yml (hunk 4 above)
Comment (SViccari) [reply circle.yml:17]: Out of curiosity, is there a benefit to having `CIRCLE_PR_USERNAME` set for some but not others? Would it be reasonable for the test suite to run the same way for everyone?

Comment (joshsmith) [reply circle.yml:17]: @SViccari the test suite can't run the same way for everyone. See https://circleci.com/docs/fork-pr-builds/#security-implications-of-running-builds-for-pull-requests-from-forks for discussion of security implications of leaking sensitive environment variables.

File: circle.yml (hunk 1 above)
Comment (Le6ow5k1) [reply circle.yml:17]: i think there is no need for that, hopefully PR will be merged soon

File: test/support/factories.ex (hunk 2 above)
Comment (Le6ow5k1) [reply test/support/factories.ex:135]: ~~you're right only markdown is needed~~

**Review by joshsmith — APPROVED** [review]

**Comment by joshsmith** [comment]
@Le6ow5k1 just squash and rebase when you have a moment, then we'll merge. Thanks!

**Comment by Le6ow5k1** [comment]
@JoshSmith 📦

**Comment by joshsmith** [comment]
❤️ 🎈 🎉

**Pull request merged**